### ISSUE TITLE
Cannot compile the source code under Win8.1 + VS2013 Update 5

### ISSUE THREAD

**Issue by xqyd** [Microsoft/CNTK#32]
Hi, I am new to CNTK, so I followed the step-by-step setup on https://github.com/Microsoft/CNTK/wiki/Setup-CNTK-on-Windows
I've download all the third party files and set the environmental variables, but when I open the CNTKSolution and build the CNTK project, I got the following errors:

1>------ Rebuild All started: Project: MathCUDA, Configuration: Debug x64 ------
2>------ Rebuild All started: Project: SequenceTrainingLib, Configuration: Debug x64 ------
3>------ Rebuild All started: Project: EvalWrapper, Configuration: Debug x64 ------
3>  wrapper.cpp
1>  
1>  D:\2015erdos\Install CNTK\CNTK-master\CNTK-master\Source\Math>"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc.exe" -ccbin "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\x86_amd64"  -I..\Common\include\ -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\include"   -lineinfo  --keep-dir x64\Debug -maxrregcount=0  --machine 64 --compile  -Xcudafe "--diag_suppress=field_without_dll_interface"  -g  -use_fast_math -D_DEBUG -DNO_SYNC -DWIN32 -D_WINDOWS -D_USRDLL -DMATH_EXPORTS -DUSE_CUDNN -D_UNICODE -DUNICODE -Xcompiler "/EHsc /W4 /nologo /Od /Zi /RTC1 /MDd  " -o x64\Debug\MathCUDA\GPUTensor.cu.obj "D:\2015erdos\Install CNTK\CNTK-master\CNTK-master\Source\Math\GPUTensor.cu" -clean 
2>  parallelforwardbackward.cpp
2>C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\crtdefs.h(496): error C2371: 'size_t' : redefinition; different basic types
2>          parallelforwardbackward.cpp : see declaration of 'size_t'
3>C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\crtdefs.h(496): error C2371: 'size_t' : redefinition; different basic types
3>          wrapper.cpp : see declaration of 'size_t'
3>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(889): error C4235: nonstandard extension used : '__asm' keyword not supported on this architecture
3>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(890): error C2065: 'mov' : undeclared identifier
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(889): error C4235: nonstandard extension used : '__asm' keyword not supported on this architecture
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(890): error C2065: 'mov' : undeclared identifier
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(890): error C2146: syntax error : missing ';' before identifier 'ecx'
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(890): error C2065: 'ecx' : undeclared identifier
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(891): error C2146: syntax error : missing ';' before identifier 'mov'
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(891): error C2065: 'mov' : undeclared identifier
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(891): error C2146: syntax error : missing ';' before identifier 'eax'
2>C:\Program Files (x86)\Windows Kits\8.1\Include\um\winnt.h(891): error C2065: 'eax' : undeclared identifier
......

GitHub won't let me to paste the full output of the compiling results. So I attached it as a txt file.
[Compiling errors.txt](https://github.com/Microsoft/CNTK/files/106086/Compiling.errors.txt)

Any idea how to fix it? Thanks!!!

**Comment by eldakms** [comment]
Hi xqyd,

We could not reproduce this on our machine (also VS2013 update5, Win 8.1). It seems that the your environment is somehow broken (usually the 'size_t' problem indicates x86/64 bit problems).

Could you provide more information regarding your environment (compiler version, PATH)? Also if you go to the "Configuration Properties -> C/C++ -> General" of SequenceTrainingLib project, and change the "Supress Startup Banner" to "No", and try to rebuild, what is the output?

Thanks!

**Comment by xqyd** [comment]
Hi eldakms,
I figured this out by find that the vs2010 and vs2013 has mixed c1.dll and c2.dll. After uninstall vs2010, the 'size_t' problem was gone!
And the SequenceTrainingLib can be build successfully.
However, I am still having difficulties in compiling the MathCUDA project, the error output is in the attached file. It said

"error MSB3721: The command ""C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc.exe" -gencode=arch=compute_20,code=\"compute_20,compute_20\" --use-local-env --cl-version 2013 -ccbin "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\x86_amd64"  -I..\Common\include\ -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\include"   -lineinfo  --keep-dir x64\Debug -maxrregcount=0  --machine 64 --compile -cudart static -Xcudafe "--diag_suppress=field_without_dll_interface"  -g  -use_fast_math -D_DEBUG -DNO_SYNC -DWIN32 -D_WINDOWS -D_USRDLL -DMATH_EXPORTS -DUSE_CUDNN -D_UNICODE -DUNICODE -Xcompiler "/EHsc /W4 /nologo /Od /Zi /RTC1 /MDd  /WX" -o x64\Debug\MathCUDA\GPUTensor.cu.obj "D:\2015erdos\Install CNTK\CNTK-master\CNTK-master\Source\Math\GPUTensor.cu"" exited with code 2."

Any idea what's this for?
Thanks a lot!!!
[Compiling errors_CUDA.txt](https://github.com/Microsoft/CNTK/files/106905/Compiling.errors_CUDA.txt)

**Comment by eldakms** [comment]
It seems you have some invalid characters in your Cuda installation (see the warning):
1>c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\math_functions.h : error C2220: warning treated as error - no 'object' file generated
1>c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\math_functions.h : warning C4819: The file contains a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss

Thank you!

**Comment by xqyd** [comment]
Hi eldakms,
Thanks for the hint. Although I did not touch the cuda installation, it is found that the system locale have to be English for CNTK. I was using Chinese locale for my Win8.1 before.

However, after changing the system locale, I've run into another problem that the xcopy does not work properly. And the reason is that it does not support long path, so after I copied the whole solution to the C:\CNTK-master, everything starts to work!!!

So here're the lessons I've learned in compiling CNTK under win8.1+vs2013 update 5:
1. Set the system locale to English.
2. Use short path to save the source code/solution files.
3. Only VS2013 can work well with the CNTK, do not mix with other version of visual studios.
